src/server/pages/user-page.jsx

```diff
@@ -50,6 +50,8 @@ export const isLoggedIn = userInfo => "name" in userInfo;
 export default class UserPage extends React.Component {
   render() {
     const { thisUser, userInfo, notebookList } = this.props;
```

Comment:
I've been very sloppy with these server components as I've added them, including a bit of logic in the render step, but hope we can apply the same code quality on these pages as we move forward. Is there any chance we can use this as an excuse to pull the logic out of `render` and put it in a function?

Reply:
Sure, what did you have in mind? A class-specific helper function or... ?

Reply:
As discussed, let's just merge this as-is. We can refactor this later.